Merge DNS search tables between IPv4 and IPv6

Without this patch the IPv4 search tables will be overwritten
once an IPv6 domain is added. With this patch both tables
will be merged.

Signed-off-by: Bart vdr. Meulen <bartvdrmeulen@gmail.com>
Signed-off-by: Remy Bohmer <linux@bohmer.net>
---
 Port-linux/lowlevel-options-linux.c |   13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

Index: dibbler-0.7.3/Port-linux/lowlevel-options-linux.c
===================================================================
--- dibbler-0.7.3.orig/Port-linux/lowlevel-options-linux.c
+++ dibbler-0.7.3/Port-linux/lowlevel-options-linux.c
@@ -199,11 +199,14 @@ int domain_add(const char* ifname, int i
 	return LOWLEVEL_ERROR_FILE;
     while (fgets(buf,511,f)) {
 	if ( (!found) && (strstr(buf, "search")) ) {
-	    if (strlen(buf))
-		buf[strlen(buf)-1]=0;
-	    fprintf(f2, "%s %s\n", buf, domain);
-	    found = 1;
-	    continue;
+	    if (!strstr(buf,domain)) {
+		if (strlen(buf))
+		    buf[strlen(buf)-1]=0;
+	        fprintf(f2, "%s %s\n", buf, domain);
+		found = 1;
+		continue;
+	    }
+
 	}
 	fprintf(f2,"%s",buf);
     }
